crates/sui-types/src/temporary_store.rs

```diff
@@ -1494,7 +1501,14 @@ impl<S: ChildObjectResolver> Storage for TemporaryStore<S> {
 
 impl<S: BackingPackageStore> BackingPackageStore for TemporaryStore<S> {
     fn get_package_object(&self, package_id: &ObjectID) -> SuiResult<Option<Object>> {
-        self.store.get_package_object(package_id)
+        self.store.get_package_object(package_id).map(|obj| {
+            // Track object but leave unchanged
+            if let Some(v) = obj.clone() {
+                // Can this lock ever block execution?
+                self.runtime_read_objects.write().insert(*package_id, v);
```

Comment:
Are `runtime_read_objects` inserted elsewhere?

Reply:
Not always, especially for Move calls.